evm/vm/evm.py

```diff
@@ -243,30 +236,42 @@ def import_block(self, block):
         except BlockNotFound:
             raise ValidationError("Parent ({0}) of block {1} not found".format(
                 block.header.parent_hash, block.header.hash))
+
         init_header = self.create_header_from_parent(parent_header)
         parent_evm = type(self)(self.db, init_header)
         imported_block = parent_evm.get_vm().import_block(block)
```

Comment:
Yeah, that is a nice improvement, but it can't be a classmethod since we need access to instance variables (e.g. self.db). Also, the lines above 239 probably belong in the new method as well since they look up the current block's parent.